app/views/books/show.html.erb

```diff
@@ -27,29 +27,76 @@
           <h1 class="card-title"><%= @book.title %></h1>
           <p class="card-text"><%= @book.description %></p>
           <p class="card-text"><strong>Creator:</strong> <%= @user.email %></p>
-          <%= link_to 'Back to Books', root_path, class: "btn btn-primary" %>
-          <%= link_to 'Delete', @book, method: :delete, data: { confirm: 'Are you sure?', "turbo-method": :delete }, class: 'btn btn-danger ms-2' %>
+          <p class="card-text"><strong>Average Rating:</strong> <%= @book.average_rating %> / 5</p>
+
+          <div>
+            <!-- Rate Button -->
+            <%= link_to "Rate this Book", "#", class: "btn btn-secondary", data: { bs_toggle: "modal", bs_target: "#rateBookModal" } %>
+          </div>
+
+          <%= link_to "Back to Books", root_path, class: "btn btn-primary mt-2" %>
+          <%= link_to "Delete", @book, method: :delete, data: { confirm: "Are you sure?", "turbo-method": :delete }, class: "btn btn-danger ms-2 mt-2" %>
+        </div>
+      </div>
+
+      <!-- Rating Modal -->
+      <div class="modal fade" id="rateBookModal" tabindex="-1" aria-labelledby="rateBookModalLabel" aria-hidden="true">
+        <div class="modal-dialog">
+          <div class="modal-content">
+            <div class="modal-header">
+              <h5 class="modal-title" id="rateBookModalLabel">Rate <%= @book.title %></h5>
+              <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
+            </div>
+            <div class="modal-body">
+          <% if @rating.new_record? %>
+            <!-- Creating a new rating -->
+            <%= form_with(model: [@book, @rating], url: book_ratings_path(@book), method: :post, local: true) do |form| %>
+              <div class="form-group">
+                <%= form.label :rating, "Your Rating" %>
+                <%= form.select :rating, options_for_select(1..5), {}, class: "form-select" %>
+              </div>
+              <div class="modal-footer">
+                <%= form.submit "Submit Rating", class: "btn btn-primary" %>
+                <button type="button" class="btn btn-secondary" data-bs-dismiss="modal">Close</button>
+              </div>
+            <% end %>
+          <% else %>
+            <!-- Updating an existing rating -->
+            <%= form_with(model: [@book, @rating], url: book_rating_path(@book, @rating), method: :patch, local: true) do |form| %>
+              <div class="form-group">
+                <%= form.label :rating, "Your Rating" %>
+                <%= form.select :rating, options_for_select(1..5, selected: @rating.rating), {}, class: "form-select" %>
```

Comment:
CREATE TABLE table (
    id INTEGER PRIMARY KEY,
    value_column INTEGER CHECK (value_column <= 4)
);

Add migration check like this, so no one can put more rating in sql